uvacw/osd2f
# Feature/dot name support

# Support for global keys that include '.' 

## Closes #96 
____

* Separated js files for easier testing
* created tests for file and object parsing using jest
* created testcases for files that include a `.` in the global key
* implemented JSON parsing using `jsonPath` 
* created sample generator for short messages with a `.` in global key
* generated new sample data 

## Assumptions

* we expect `.` only in the global keys, not in object keys

## Usage / Minimal Example

![osd2f_upload_with_dot_in_global_name](https://user-images.githubusercontent.com/1457643/168313096-04f3c16b-7a9f-4227-9dff-b07a43b84a14.gif)


Instructions on how to verify changes this PR by running code


1. See the new [default config](https://github.com/uvacw/osd2f/blob/feature/dot-name-support/osd2f/settings/default_upload_settings.yaml)

2. Start a server 
```bash
OSD2F_SECRET=secret \
OSD2F_BASIC_AUTH="user;pass" \
OSD2F_MODE=Development \
osd2f -m Development
```
3. upload the new sample content
4. investigate the `messages.json` result

## Checklist
- [X] Added tests if appropriate (and it should always be)
- [X] Created new issues when required
